File: app/src/main/java/me/tagavari/airmessage/helper/ColorHelper.java
package me.tagavari.airmessage.helper;

import android.content.res.Resources;

import me.tagavari.airmessage.R;
import me.tagavari.airmessage.enums.ServiceHandler;
import me.tagavari.airmessage.enums.ServiceType;

public class ColorHelper {
	/**
	 * Gets the color used to represent the specified service
	 */
	public static int getServiceColor(Resources resources, @ServiceHandler int serviceHandler, @ServiceType String serviceType) {
		//AirMessage bridge
		if(serviceHandler == ServiceHandler.appleBridge) {
			//Returning a default color if the service is invalid
			if(serviceType == null) return resources.getColor(R.color.colorMessageDefault, null);
			
			switch(serviceType) {
				case ServiceType.appleMessage:
					//iMessage
					return resources.getColor(R.color.colorPrimary, null);
				case ServiceType.appleSMS:
					//SMS bridge
					return resources.getColor(R.color.colorMessageTextMessageForwarding, null);
				default:
					return resources.getColor(R.color.colorMessageDefault, null);
			}
		}
		//System messaging
		else if(serviceHandler == ServiceHandler.systemMessaging) {
			switch(serviceType) {
				case ServiceType.systemSMS:
					return resources.getColor(R.color.colorMessageTextMessage, null);
				case ServiceType.systemRCS:
					return resources.getColor(R.color.colorMessageRCS, null);
				default:
					return resources.getColor(R.color.colorMessageDefault, null);
			}
		}
		
		//Returning a default color
		return resources.getColor(R.color.colorMessageDefault, null);
	}
}

File: common/me/tagavari/airmessage/common/helper/ColorHelper.kt
package me.tagavari.airmessage.common.helper

import android.content.res.Resources
import me.tagavari.airmessage.R
import me.tagavari.airmessage.common.enums.ServiceHandler
import me.tagavari.airmessage.common.enums.ServiceType

object ColorHelper {
	/**
	 * Gets the color used to represent the specified service
	 */
	@JvmStatic
	fun getServiceColor(resources: Resources, @ServiceHandler serviceHandler: Int, @ServiceType serviceType: String?): Int {
		//AirMessage bridge
		if(serviceHandler == ServiceHandler.appleBridge) {
			//Returning a default color if the service is invalid
			return when(serviceType) {
				ServiceType.appleMessage -> resources.getColor(R.color.colorPrimary, null) //iMessage
				ServiceType.appleSMS -> resources.getColor(R.color.colorMessageTextMessageForwarding, null) //SMS bridge
				else -> resources.getColor(R.color.colorMessageDefault, null)
			}
		} else if(serviceHandler == ServiceHandler.systemMessaging) {
			return when(serviceType) {
				ServiceType.systemSMS -> resources.getColor(R.color.colorMessageTextMessage, null)
				ServiceType.systemRCS -> resources.getColor(R.color.colorMessageRCS, null)
				else -> resources.getColor(R.color.colorMessageDefault, null)
			}
		}
		
		//Returning a default color
		return resources.getColor(R.color.colorMessageDefault, null)
	}
}